mailserver/leveldb_impl.go

```diff
@@ -0,0 +1,113 @@
+package mailserver
```

Comment:
I would call it `mailserver_db_leveldb.go` and `mailserver_db_postgres.go` instead.